File: 2091-removing-minimum-and-maximum-from-array/2091-removing-minimum-and-maximum-from-array.java
class Solution {
    public int minimumDeletions(int[] nums) {

        int n = nums.length;

        int min = nums[0];
        int max = nums[0];

        int minIndex = 0;
        int maxIndex = 0;
        for (int i = 1; i < n; i++) {

            if (nums[i] < min) {
                min = nums[i];
                minIndex = i;
            }

            if (nums[i] > max) {
                max = nums[i];
                maxIndex = i;
            }
        }

        int a = Math.min(minIndex, maxIndex);
        int b = Math.max(minIndex, maxIndex);

        int front = b + 1;

        int back = n - a;
        int both = (a + 1) + (n - b);

        return Math.min(front, Math.min(back, both));
    }
}

File: 2091-removing-minimum-and-maximum-from-array/2091-removing-minimum-and-maximum-from-array.py
class Solution:
    def minimumDeletions(self, nums: List[int]) -> int:
        n = len(nums)

        minnum = nums[0]
        maxnum = nums[0]

        minIndex = 0
        maxIndex = 0

        for i in range(1, n):
            if nums[i] < minnum:
                minnum = nums[i]
                minIndex = i

            if nums[i] > maxnum:
                maxnum = nums[i]
                maxIndex = i

        a = min(minIndex, maxIndex)
        b = max(minIndex, maxIndex)

        front = b + 1

        back = n - a

        both = (a + 1) + (n - b)

        return min(front, min(back,both))